cmd/kubectl-request/kubectl-request.go

```diff
@@ -202,12 +202,17 @@ func (ac *accessCommand) Request(cmd *cobra.Command, args []string) error {
 
 	}
 
-	// default to first container if none is set (like `kubectl exec`)
+	// default to default or first container if none is set (like `kubectl exec`)
 	if ac.execOptions.Container == "" {
 		if len(selectedPod.Spec.Containers) == 0 {
 			return fmt.Errorf("no containers in pod %q", selectedPod.Name)
 		}
-		ac.execOptions.Container = selectedPod.Spec.Containers[0].Name
+
+		if selectedPod.ObjectMeta.Annotations["kubectl.kubernetes.io/default-container"] != "" {
```

Comment:
Just an idea here, what about doing a block like this:

```go
if val := selectedPod.ObjectMeta.Annotations["kubectl.kubernetes.io/default-container"]; val != "" {
  ac.execOptions.Container = val
} else {
  ac.execOptions.Container = selectedPod.Spec.Containers[0].Name
}
```

It might look a bit cleaner in my opinion.

Reply:
I don't really like those assignments in an if-statement, but I could assign it before the if-statement if you prefer to have a variable for the value of the annotation.

Reply:
No worries. It was just an idea :smile: . In such case, let's merge it as is